src/runtime_src/core/common/ishim.h

```diff
@@ -335,52 +335,52 @@ struct shim : public DeviceType
       throw system_error(ret, "failed to close ip context");
   }
 
-  xclBufferHandle
+  xrt_buffer_handle
   alloc_bo(size_t size, unsigned int flags) override
   {
     auto bo = xclAllocBO(DeviceType::get_device_handle(), size, 0, flags);
     if (bo == XRT_NULL_BO)
       throw std::bad_alloc();
 
-    return bo;
+    return to_xrt_buffer_handle(bo);
   }
 
-  xclBufferHandle
+  xrt_buffer_handle
   alloc_bo(void* userptr, size_t size, unsigned int flags) override
   {
     auto bo = xclAllocUserPtrBO(DeviceType::get_device_handle(), userptr, size, flags);
     if (bo == XRT_NULL_BO)
       throw std::bad_alloc();
 
-    return bo;
+    return to_xrt_buffer_handle(bo);
   }
 
   void
-  free_bo(xclBufferHandle bo) override
+  free_bo(xrt_buffer_handle bo) override
   {
-    xclFreeBO(DeviceType::get_device_handle(), bo);
+    xclFreeBO(DeviceType::get_device_handle(), to_xclBufferHandle(bo));
   }
 
   xclBufferExportHandle
-  export_bo(xclBufferHandle bo) const override
+  export_bo(xrt_buffer_handle bo) const override
   {
-    auto ehdl = xclExportBO(DeviceType::get_device_handle(), bo);
+    auto ehdl = xclExportBO(DeviceType::get_device_handle(), to_xclBufferHandle(bo));
     if (ehdl == XRT_NULL_BO_EXPORT)
       throw system_error(EINVAL, "Unable to export BO");
     if (ehdl < 0) // system error code
       throw system_error(ENODEV, "Unable to export BO");
```

Comment:
This error is different from line 369, yet the message is the same.  One might wish to update the message to be more relevant.

Reply:
Fixed both line 369 and 380 to mention bad handle.